demo-project/src/docker_requirements.txt

```
@@ -1,4 +1,4 @@
-kedro~=0.18.0
+kedro>=0.18.0
 kedro-datasets[pandas.CSVDataSet,pandas.ExcelDataSet, pandas.ParquetDataSet, plotly.PlotlyDataSet, matplotlib.MatplotlibWriter]~=1.0
```

I would try to use `kedro-datasets` 2.0.0 if possible for the demo project so it's on the very latest release. 

```suggestion
kedro-datasets[pandas.CSVDataset,pandas.ExcelDataset, pandas.ParquetDataset, plotly.PlotlyDataset, matplotlib.MatplotlibWriter]~=2.0
```